packages/eslint-plugin/src/rules/restrict-plus-operands.ts

```diff
@@ -72,19 +72,24 @@ export default util.createRule<Options, MessageIds>({
   },
   defaultOptions: [
     {
+      allowAny: true,
+      allowBoolean: true,
+      allowNullish: true,
+      allowNumberAndString: true,
+      allowRegExp: true,
```

Comment:
@JoshuaKGoldberg i don't remember our discussion here - why did we turn all of the options on?
I thought we discussed that turning all the rule options on was bad as it made the rule effectively useless.

Reply:
Your example is template strings - but this is restrict-plus-operands.

So was this an accidental mix-up?

Reply:
I think it applies to both of them. I've seen both happen with users. Which is why I keep mixing them up in comments 😅 to me, they're similar -just not exactly the same- cases.

Reply:
I guess the question is - why even have these turned on at all if all the options are on by default?
The rules enforce pretty nothing like this so it's just wasted lint time.

[Eg for RPO there's the following cases](https://typescript-eslint.io/play/#ts=5.1.6&fileType=.tsx&code=CYUwxgNghgTiAEYD2A7AzgF3pmAubGMAligOYDcAUKJLAsuligK4C2%2BLrARiDFTdDiJUmAjAByzCBA5SI8AD5iSFauEH0RTNpOmzpi%2BJx581tIQ1FckSGfGu2QUFP3V1hjeEi4ArfAG8AX1dzTU9nAE98SJCND1FmFABrFCQAdxR8RJT0l0pKHHgAajEqTmKjNioCwgrOavKSzl0IMrYKnBaqQpLOuW7app1%2B-Mb7G1aamAqHSdmZiYb2ku8fcngAeg34GAAHJHgMAAsoLCQAN15oXbR4ABUAZXheGCQYKYrV9a2d-dHl%2BAxD4lIGUMbZVIZb7bPYHY6nLyXGDXW6PZ4wV7vHrwCG5aG-JCUIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6aFfKsu4%2BWZLQD2paAEMmAE1QZIiaNCHRI4MAF8QaoA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false). Of these 16 cases the default options mean the rule only matches FOUR cases - two of which are TS already errors.

So we're recommending a rule to everyone to stop them from doing: `string + object` and `string + unknown`. The former is also banned under `no-base-to-string`, and one could argue the latter should be allowed under the `allowAny` option.

tl;dr - why are we even turning on the rules in the recommended set if they don't do anything with an overly permissive default option?

Reply:
A couple reasons swayed me:

* (lesser) There is benefit to `string + unknown`
* (greater) I've seen a projects turn off `no-base-to-string` without turning off `restrict-plus-operands`

From a performance perspective, if one of these two rules is already asking for the types of these, I some level of caching on the TypeScript side would make the other rule not a significant change.

Reply:
The thing I'd point to is that we added this big caution block to the RPO docs because we didn't think people using the options was good - and yet it's the default that everyone uses these options?
https://typescript-eslint.io/rules/restrict-plus-operands#:~:text=%2C%0A%5D%3B-,CAUTION,-We%20generally%20recommend

Just seems like we're contradicting ourselves "don't use these options but also they're all turned on by default"

Reply:
Yeah :/ agreed. I don't like this.

Reply:
Personal preference:
Given this rule is really doing nothing and a lot of people don't like it - we should just undo this change and only include it in `strict-type-checked`